Add required ciphersuites for connectivity to Azure Front Door (#5323)
MicrosoftDocs/microsoft-365-docs: microsoft-365/compliance/technical-reference-details-about-encryption.md

Office 365 responds to a connection request by first attempting to connect using the most secure cipher suite. If the connection doesn't work, Office 365 tries the second most secure cipher suite in the list, and so on. The service continues down the list until the connection is accepted. Likewise, when Office 365 requests a connection, the receiving service chooses whether TLS will be used and which cipher suite to use.

> Certain Office 365 products (including Microsoft Teams) use [Azure Front Door](https://docs.microsoft.com/en-us/azure/frontdoor/front-door-overview) to terminate TLS connections and route network traffic efficiently. One of the [ciphersuites supported by Azure Front Door over TLS 1.2](https://docs.microsoft.com/en-us/azure/frontdoor/front-door-faq#what-are-the-current-cipher-suites-supported-by-azure-front-door-) must be enabled to successfully connect to these products. Office recommends enabling one of the two ECDHE ciphersuites marked "Required for AFD" in the table below for the most security. 

Reviewer: ```suggestion
Certain Office 365 products (including Microsoft Teams) use [Azure Front Door](/azure/frontdoor/front-door-overview) to terminate TLS connections and route network traffic efficiently. One of the [cipher suites supported by Azure Front Door over TLS 1.2](/azure/frontdoor/front-door-faq#what-are-the-current-cipher-suites-supported-by-azure-front-door-) must be enabled to successfully connect to these products. We recommend enabling one of the two ECDHE cipher suites marked "Required for AFD" in the table below for maximum security. 
```

| Cipher suite | Key exchange algorithm/strength | Forward Secrecy | Cipher/strength | Authentication algorithm | Required for AFD |
|:-----|:-----|:-----|:-----|:-----|:-----|
| TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384  <br/> | ECDH/192  <br/> | Yes  <br/> | AES/256  <br/> | RSA/112  <br/> | Yes  <br/> |
| TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256  <br/> | ECDH/128  <br/> | Yes  <br/> | AES/128  <br/> | RSA/112  <br/> | Yes  <br/> |
| TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384  <br/> | ECDH/192  <br/> | Yes  <br/> | AES/256  <br/> | RSA/112  <br/> | No   <br/> |
| TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256  <br/> | ECDH/128  <br/> | Yes  <br/> | AES/128  <br/> | RSA/112  <br/> | No   <br/> |
| TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA     <br/> | ECDH/192  <br/> | Yes  <br/> | AES/256  <br/> | RSA/112  <br/> | No   <br/> |
| TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA     <br/> | ECDH/128  <br/> | Yes  <br/> | AES/128  <br/> | RSA/112  <br/> | No   <br/> |
| TLS_RSA_WITH_AES_256_GCM_SHA384        <br/> | RSA/112   <br/> | No   <br/> | AES/256  <br/> | RSA/112  <br/> | No   <br/> |
| TLS_RSA_WITH_AES_128_GCM_SHA256        <br/> | RSA/112   <br/> | No   <br/> | AES/256  <br/> | RSA/112  <br/> | No   <br/> |

> 

Reviewer 2: Suggest indicating below 2 ciphers are supported by AFD for for Windows 7 and 8 compatibility.
TLS_DHE_RSA_WITH_AES_256_GCM_SHA384 (0x9f)   DH 2048 bits
TLS_DHE_RSA_WITH_AES_128_GCM_SHA256 (0x9e)   DH 2048 bits

Author: You'd like me to make that comment in the AFD documentation? We don't have any discussion of those two cipher suites in this doc. We're documenting the non-FS `TLS_RSA` but not the `TLS_DHE_RSA`.

Author: Discussed offline with Fasheng. My mistake was thinking Win 7+ supported EDCHE. Only Win 10+ supports ECDHE and we need to clarify our recommendation that any of the four cipher suites may need to be enabled depending on the OS version.
